docs/source/corev_hw_loop.rst

```diff
@@ -43,76 +43,84 @@ Hardware Loop constraints
 
 The HWLoop constraints are:
 
--  Start and End address of an HWLoop must be word aligned
+-  Start and End address of an HWLoop must be word aligned.
+
+-  End address of an HWLoop must point to the instruction just after the last one of the loop body.
 
 -  HWLoop body must contain at least 3 instructions.
-   An illegal exception is raised otherwise.
+   An illegal instruction exception is raised otherwise.
 
 -  No Compressed instructions (RVC) allowed in the HWLoop body.
-   An illegal exception is raised otherwise.
+   An illegal instruction exception is raised otherwise.
```

Comment:
shall we say this happens when fetching the first time a compressed instruction inside the HW loop?